src/cursor/cur_index.c

```diff
@@ -347,6 +354,43 @@ __curindex_search_near(WT_CURSOR *cursor, int *exact)
     API_END_RET(session, ret);
 }
 
+/*
+ * __increment_bound_array --
+ *     Increment the given buffer by one bit. If the all of the values inside the buffer is maxed
+ *     values
+ */
+static inline int
+__increment_bound_array(WT_ITEM *user_item)
+{
+    size_t usz;
+    uint8_t *userp;
+    int i;
+
+    usz = user_item->size;
+    userp = (uint8_t *)user_item->data;
+
+    if (usz <= 0)
+        return (0);
+
+    /*
+     * First loop through all max values on the buffer from the end. This is to find the appropriate
+     * position to increment add one to the byte.
+     */
+    for (i = usz - 1; i >= 0 && userp[i] == UINT8_MAX; --i)
+        ;
+
+    /*
+     * If all of the buffer are max values, we don't need to do anything as the key format is a
+     * fixed length format. Ideally we double check that the table format has a fixed length string.
+     */
+    if (i < 0)
+        return (0);
```

Comment:
@ddanderson do you think we should assert that the key format has to be fixed length here?